console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,90 @@ type ConsolePluginSpec struct {
 	// i18n is the configuration of plugin's localization resources.
 	// +optional
 	I18n ConsolePluginI18n `json:"i18n"`
+	// contentSecurityPolicy is a list of Content Security Policy directives for the plugin.
+	// Each directive specifies a list of values that indicate server origins and script endpoints
+	// from which the plugin's assets can be loaded. This helps guard against cross-site
+	// scripting (XSS) attacks by enforcing strict security policies for asset loading.
+	// Dynamic plugins should to specify this field if they are loading assets form outside
+	// the cluster or if violation reports are observed.
+	// CSP violation reports can be viewed in browser's console during development and testing
+	//  of the plugin in the OpenShift web console.
+	// Available directives are default-src, script-src, img-src, style-src and font-src.
+	// Each of the available CSP directive may be defined only once in the list.
+	// By default the console server adds the value 'self'to all the various 'src' directives.
```

Comment:
```suggestion
	// By default the console server adds the value 'self' to all directives.
```
Reading this I'd wonder by `self` is added? And you say by default, does this mean I can disable this behaviour?

Reply:
Console needs `self` to function for all of these values. It can't be removed.

All of the plugin customizations append to the default console values that console needs to work.

Reply:
Note that in the future we could add directives that don't specify sources (although right now I think everything we expose is a "src" directive). We'd have to come back and update the text here later if we leave that out.

Reply:
So that is to say, if this were a directive that were not a `src` style, then it wouldn't also need the `self` prepending?

Reply:
Right, `self` only makes sense as a value for the various CSP `src` directives.